src/main/java/com/ebivariation/contigalias/controller/ContigAliasController.java

```diff
@@ -48,37 +49,39 @@ public ContigAliasController(AssemblyService assemblyService, ChromosomeService
+                                                                               createPageRequest(page, size));
+        return createAppropriateResponseEntity(entities);
     }
 
     @ApiOperation(value = "Get an assembly using its Genbank accession.")
     @GetMapping(value = "v1/assemblies/genbank/{genbank}", produces = "application/json")
-    public ResponseEntity<AssemblyEntity> getAssemblyByGenbank(@PathVariable String genbank) {
-        Optional<AssemblyEntity> entity = assemblyService.getAssemblyByGenbank(genbank);
-        return entity.map(assemblyEntity -> new ResponseEntity<>(assemblyEntity, HttpStatus.OK))
-                     .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    public ResponseEntity<List<AssemblyEntity>> getAssemblyByGenbank(@PathVariable String genbank,
+                                                                     @RequestParam(required = false) Integer page,
+                                                                     @RequestParam(required = false) Integer size) {
+        List<AssemblyEntity> entities = assemblyService.getAssemblyByGenbank(genbank, createPageRequest(page, size));
+        return createAppropriateResponseEntity(entities);
     }
 
     @ApiOperation(value = "Get an assembly using its Refseq accession.")
     @GetMapping(value = "v1/assemblies/refseq/{refseq}", produces = "application/json")
-    public ResponseEntity<AssemblyEntity> getAssemblyByRefseq(@PathVariable String refseq) {
-        Optional<AssemblyEntity> entity = assemblyService.getAssemblyByRefseq(refseq);
-        return entity.map(assemblyEntity -> new ResponseEntity<>(assemblyEntity, HttpStatus.OK))
-                     .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    public ResponseEntity<List<AssemblyEntity>> getAssemblyByRefseq(@PathVariable String refseq,
+                                                                    @RequestParam(required = false) Integer page,
+                                                                    @RequestParam(required = false) Integer size) {
+        List<AssemblyEntity> entities = assemblyService.getAssemblyByRefseq(refseq, createPageRequest(page, size));
+        return createAppropriateResponseEntity(entities);
     }
 
     @ApiOperation(value = "Get an assembly using its Taxonomic ID.")
     @GetMapping(value = "v1/assemblies/taxid/{taxid}", produces = "application/json")
-    public ResponseEntity<List<AssemblyEntity>> getAssembliesByTaxid(@PathVariable long taxid) {
-        List<AssemblyEntity> entities = assemblyService.getAssembliesByTaxid(taxid);
-        if (entities != null && !entities.isEmpty()) {
-            return new ResponseEntity<>(entities, HttpStatus.OK);
-        } else {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
+    public ResponseEntity<List<AssemblyEntity>> getAssembliesByTaxid(@PathVariable long taxid,
```

Comment:
the problem with this pagination approach is that when users do a query like "/assemblies/taxid/9606" and receive a list with 10 results, they might not know about the pagination thing and think that's the complete result.

We have to provide something in the result to signal that it's only a page. I'm not a huge fan of HATEOAS but it seems to be the closest to a standard that we currently have. take a look at one of our endpoints:
- swagger: https://www.ebi.ac.uk/eva/webservices/rest/swagger-ui.html#!/studies/getBrowsableStudiesUsingGET_1
- single page example: https://www.ebi.ac.uk/eva/webservices/rest/v2/studies?species=ecaballus&assembly=20&pageNumber=0&pageSize=20
- several page example: https://www.ebi.ac.uk/eva/webservices/rest/v2/studies?species=hsapiens&assembly=grch37&pageNumber=0&pageSize=20
- code: https://github.com/EBIvariation/eva-ws/blob/c4a5bbf892cbe2aa46c2f21f8f3fb5a4f3ed99b2/eva-server/src/main/java/uk/ac/ebi/eva/server/ws/StudyWSServerV2.java#L85

The good thing about it is that it tells you how many pages are there, and also gives you the links to get the next/previous/first/last pages (when it makes sense).